packages/@aws-cdk/aws-appmesh/lib/http-route-path-match.ts

```diff
@@ -0,0 +1,247 @@
+    return new HttpRoutePathSpecificMatch({ exact: path });
+  }
+
+  /**
+   * The value of the path must match the specified regex.
+   * @param regex The regex used to match the path.
+   */
+  public static regex(regex: string): HttpRoutePathMatch {
+    return new HttpRoutePathSpecificMatch({ regex: regex });
+  }
+
+  /**
+   * The value of the path must match the specified prefix.
+   *
+   * @param prefix This parameter must always start with /, which by itself matches all requests to the virtual service name.
+   *  You can also match for path-based routing of requests. For example, if your virtual service name is my-service.local
+   *  and you want the route to match requests to my-service.local/metrics, your prefix should be /metrics.
+   */
+  public static startingWith(prefix: string): HttpRoutePathMatch {
+    return new HttpRoutePrefixMatchImpl(prefix);
+  }
+
+  /**
+   * Returns the route path match configuration.
+   */
+  public abstract bind(scope: Construct): HttpRoutePathMatchConfig;
+}
+
+class HttpRoutePrefixMatchImpl extends HttpRoutePathMatch {
+  constructor(
+    private readonly prefix: string,
+  ) {
+    super();
+  }
+
+  bind(_scope: Construct): HttpRoutePathMatchConfig {
+    return {
+      prefixPathMatch: this.prefix,
+    };
+  }
+}
+
+class HttpRoutePathSpecificMatch extends HttpRoutePathMatch {
+  constructor(
+    private readonly match: CfnRoute.HttpPathMatchProperty,
+  ) {
+    super();
+  }
+
+  bind(_scope: Construct): HttpRoutePathMatchConfig {
+    return {
+      wholePathMatch: this.match,
+    };
+  }
+}
+
+/**
+ * Path and Prefix properties for HTTP gateway route match and rewrite.
+ */
+export interface HttpGatewayRoutePathMatchConfig {
+  /**
+   * Gateway route CFN configuration for HTTP path match.
+   *
+   * @default - no path match.
+   */
+  readonly wholePathMatch?: CfnGatewayRoute.HttpPathMatchProperty;
+
+  /**
+   * String defining the HTTP prefix match.
+   *
+   * @default - matches requests with any path
+   */
+  readonly prefixPathMatch?: string;
+
+  /**
+   * Gateway route CFN configuration for HTTP path rewrite.
+   *
+   * @default - no path rewrite
+   */
+  readonly wholePathRewrite?: CfnGatewayRoute.HttpGatewayRoutePathRewriteProperty;
+
+  /**
+   * Gateway route CFN configuration for HTTP prefix rewrite.
+   *
+   * @default - rewrite prefix to '/'.
+   */
+  readonly prefixPathRewrite?: CfnGatewayRoute.HttpGatewayRoutePrefixRewriteProperty;
+}
+
+/**
+ * Defines HTTP gateway route path or prefix request match.
+ */
+export abstract class HttpGatewayRoutePathMatch {
+  /**
+   * The value of the path must match the specified value exactly.
+   *
+   * @param pathMatch The exact path to match on
+   */
+  public static exact(pathMatch: string, exactPathRewrite?: string): HttpGatewayRoutePathMatch {
+    return new HttpGatewayRoutePathSpecificMatch({ exact: pathMatch }, exactPathRewrite);
+  }
+
+  /**
+   * The value of the path must match the specified regex.
```

Comment:
Missing empty line after this line:

```suggestion
   * The value of the path must match the specified regex.
   *
```

Reply:
Addressed